fastlane/lib/fastlane/actions/download_dsyms.rb

```diff
@@ -51,21 +52,35 @@ def self.run(params)
         UI.message(message.join(" "))
 
         app.tunes_all_build_trains(platform: platform).each do |train|
-          UI.verbose("Found train: #{train.version_string}, comparing to supplied version: #{version}")
+          message = []
+          message << "Found train (version): #{train.version_string}"
+          message << ", comparing to supplied version: #{version}" if version
+          UI.verbose(message.join(" "))
+
           if version && version != train.version_string
             UI.verbose("Version #{version} doesn't match: #{train.version_string}")
             next
           end
+
+          if min_version && min_version > Gem::Version.new(train.version_string)
```

Comment:
`min_version` is not `nil` here, even if `params[:min_version]` is `nil`. While this might still work, I assume it's not intended. If it isn't, maybe let's update line 28 to only set the `min_version` if it was actually provided by the user

Reply:
Oh right, I added the "optimisation" to only do `Gem::Version.new` once at the beginning later. Should be fixed now.